bash_completion

```diff
@@ -1266,19 +1328,20 @@ _ncpus()
 
 # Perform tilde (~) completion
 # @return  True (0) if completion needs further processing,
-#          False (> 0) if tilde is followed by a valid username, completions
-#          are put in COMPREPLY and no further processing is necessary.
+#          False (1) if tilde is followed by a valid username, completions are
+#          put in COMPREPLY and no further processing is necessary.
 _tilde()
 {
     local result=0
     if [[ ${1-} == \~* && $1 != */* ]]; then
         # Try generate ~username completions
-        COMPREPLY=($(compgen -P '~' -u -- "${1#\~}"))
-        result=${#COMPREPLY[@]}
-        # 2>/dev/null for direct invocation, e.g. in the _tilde unit test
-        ((result > 0)) && compopt -o filenames 2>/dev/null
+        if _comp_compgen COMPREPLY -P '~' -u -- "${1#\~}"; then
+            # 2>/dev/null for direct invocation, e.g. in the _tilde unit test
+            compopt -o filenames 2>/dev/null
+            return 1
+        fi
     fi
-    return "$result"
```

Comment:
eeacd68777aa993be57727726a73a7a5e451deeb cherry-picked as 177cba155a39229da986de41b2ac5bca20bf3045

Reply:
I have rebased it.